aiida/restapi/common/identifiers.py

```diff
@@ -103,16 +103,21 @@ def get_full_type_filters(full_type):
         node_type = escape_for_sql_like(node_type) + LIKE_OPERATOR_CHARACTER
         filters['node_type'] = {'like': node_type}
     else:
-        filters['node_type'] = escape_for_sql_like(node_type)
+        filters['node_type'] = {'==': node_type}
 
     if LIKE_OPERATOR_CHARACTER in process_type:
-        # Remove the trailing `LIKE_OPERATOR_CHARACTER`, escape the string and reattach the character
+        # Remove the trailing `LIKE_OPERATOR_CHARACTER`
+        # If that was the only specification: just ignore this filter (looking for any process_type)
+        # Else: escape the string and reattach the character
         process_type = process_type[:-1]
-        process_type = escape_for_sql_like(process_type) + LIKE_OPERATOR_CHARACTER
-        filters['process_type'] = {'like': process_type}
+        if process_type:
+            process_type = escape_for_sql_like(process_type) + LIKE_OPERATOR_CHARACTER
+            filters['process_type'] = {'like': process_type}
     else:
         if process_type:
-            filters['process_type'] = escape_for_sql_like(process_type)
+            filters['process_type'] = {'==': process_type}
+        else:
+            filters['process_type'] = {'or': [{'==': ''}, {'==': None}]}
```

Comment:
So, I don't think it would have the same effect because currently if `process_type` is **either** `''` or `None`, it will search for **both** `''` and `None`. Your change would make `None` just search for `None` and `''` just search for `''`.

The reason I grouped these together is that this function takes the `full_type` string (that looks like `node.type.Descriptor|process.type.descriptor`) and splits it into the `node_type` and the `process_type`. It is possible to pass an empty `process_type` in this way (`node.type.Descriptor|`) but it is not possible to do so with a null process type (as there is no simple way to distinguish between `' None'` string and `None` in `node.type.Descriptor|None`).

Reply:
so you are saying that the full types will look like `node.type|None` both when `process_type` is `''` as well as `None`? In that case, you are right that what I proposed is not the same. I am still worried about all these edge cases that really shouldn't exist and we should figure out how and why they came about. At the very least, you should add a comment here to explain that your additional clause is because there (erroneously) exist process nodes with a process_type equal to empty string or null

Reply:
> so you are saying that the full types will look like `node.type|None` both when process_type is `''` as well as `None`?

Nono, it will show `node.type|` both when it is `''` and when it is `None`. I mentioned the `node.type|None` example to illustrate that it is not easy to distinguish between these two cases because you need to encode this into a string (the `full_type`) and therefore you also need to distinguish between the null `None` and the string `'None'`.

> I am still worried about all these edge cases that really shouldn't exist and we should figure out how and why they came about. At the very least, you should add a comment here to explain that your additional clause is because there (erroneously) exist process nodes with a process_type equal to empty string or null

Ok, I can add that, and we could open an issue. I mean, it might be the case that this was a bug in some migrations that already got fixed, but nevertheless these nodes will exist in databases so I think it is not bad to be able to catch these in the most securely way possible.